vncdotool/client.py

```diff
@@ -237,11 +237,16 @@ def captureRegion(self, filename, x, y, w, h):
         log.debug('captureRegion %s', filename)
         return self._capture(filename, x, y, x+w, y+h)
 
+    def refreshScreen(self, incremental=0):
+        d = self.deferred = Deferred()
+        self.framebufferUpdateRequest(incremental=incremental)
+        d.addCallback(lambda *args: self)
```

Comment:
I think I've lost track of the changes we've been discussion.  What was the purpose of this lambda callback?

Reply:
If you don't set the callback, then call `refreshScreen` twice, you get:

```
Traceback (most recent call last):
  File "./foo.py", line 63, in main
    client.refreshScreen()
  File "/Users/agf/projects/eq/vncdotool/api.py", line 108, in proxy_call
    raise VNCDoException(result)
vncdotool.api.VNCDoException: [Failure instance: Traceback: <class 'AttributeError'>: 'Image' object has no attribute 'framebufferUpdateRequest'
```

on the second attempt because the result of the callback is used in place of `self` the second time `refreshScreen` is called. Having the callback return `self` solves the problem.

I was trying to avoid changing `commitUpdate` so this change wouldn't affect the `expectScreen` code. I've just pushed a commit that updates all of that instead (including tests). Let me know what you think.